File: src/main/java/novemberizing/ds/on/Triple.java
package novemberizing.ds.on;

import novemberizing.ds.On;

/**
 *
 * @author novemberizing, me@novemberizing.net
 * @since 2017. 1. 17.
 */
public interface Triple<A, B, C> extends On { void on(A first, B second, C third); }

File: src/main/c++/novemberizing.old.3/ds/tuple/triple.cc
#include "triple.hh"

#ifdef    __NOVEMBERIZING_DS_TUPLE__TRIPLE__EXAMPLE__

#include <iostream>

using namespace novemberizing;

int main(int argc, char ** argv) {

    ds::tuple::Triple<int, int, int> triple(1, 2, 3);

    std::cout << triple.first << ":" << triple.second << ":" << triple.third <<std::endl;

    return 0;
}

#endif // __NOVEMBERIZING_DS_TUPLE__TRIPLE__EXAMPLE__
